test/cbmc/stubs/core_sntp_stubs.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * coreSNTP v1.0.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_sntp_stubs.c
+ * @brief Definition of stubs for UDP transport and authentication interfaces of coreSNTP API.
+ */
+
+#include <stdint.h>
+#include "core_sntp_client.h"
+#include "core_sntp_serializer.h"
+#include "core_sntp_stubs.h"
+
+#define TEST_TIMESTAMP         \
+    {                          \
+        .seconds = UINT32_MAX, \
+        .fractions = 1000      \
+    }
+
+/* An exclusive bound on the times that the NetworkInterfaceSendStub will be
+ * invoked before returning a loop terminating value. This is usually defined
+ * in the Makefile of the harnessed function. */
+#ifndef MAX_NETWORK_SEND_TRIES
+    #define MAX_NETWORK_SEND_TRIES    2
+#endif
+
+/* An exclusive bound on the times that the NetworkInterfaceReceiveStub will
+ * return an unbound value. At this value and beyond, the
+ * NetworkInterfaceReceiveStub will return zero on every call. */
+#ifndef MAX_NETWORK_RECV_TRIES
+    #define MAX_NETWORK_RECV_TRIES    4
+#endif
+
+static SntpTimestamp_t testTime = TEST_TIMESTAMP;
+
+int32_t NetworkInterfaceReceiveStub( NetworkContext_t * pNetworkContext,
```

Comment:
In all of these stubs, you might want to asser the pNetworkContext is not NULL.

Reply:
I am not initializing Network context and Auth Context in my profs a this needs to be provided by the application and the way stub is written is not using network context. The stub is making sure that we send and receive bytes using the buffer provided, which I have asserted already.